code/game/machinery/porta_turret/portable_turret.dm

```diff
@@ -921,3 +921,167 @@
+		buckled_mob.pixel_y = 0
+		if(buckled_mob.client)
+			buckled_mob.client.change_view(world.view)
+	anchored = FALSE
+	. = ..()
+
+/obj/machinery/manned_turret/user_buckle_mob(mob/living/M, mob/living/carbon/user)
+	if(user.incapacitated() || !istype(user))
+		return
+	M.forceMove(get_turf(src))
+	..()
+	for(var/V in M.held_items)
+		var/obj/item/I = V
+		if(istype(I))
+			if(M.dropItemToGround(I))
+				var/obj/item/gun_control/TC = new /obj/item/gun_control(src)
+				M.put_in_hands(TC)
+		else	//Entries in the list should only ever be items or null, so if it's not an item, we can assume it's an empty hand
+			var/obj/item/gun_control/TC = new /obj/item/gun_control(src)
+			M.put_in_hands(TC)
+	M.pixel_y = 14
+	layer = 4.1
+	setDir(SOUTH)
+	playsound(src,'sound/mecha/mechmove01.ogg', 50, 1)
+	anchored = TRUE
+	if(user.client)
+		user.client.change_view(view_range)
+
+/obj/item/gun_control
+	name = "turret controls"
+	icon = 'icons/obj/weapons.dmi'
+	icon_state = "offhand"
+	w_class = WEIGHT_CLASS_HUGE
+	flags = ABSTRACT | NODROP
+	resistance_flags = FIRE_PROOF | UNACIDABLE | ACID_PROOF | NOBLUDGEON
+	var/obj/machinery/manned_turret/turret
+
+/obj/item/gun_control/New(var/obj/machinery/manned_turret/MT)
+	turret = MT
+
+/obj/item/gun_control/CanItemAutoclick()
+	return 1
+
+/obj/item/gun_control/afterattack(atom/targeted_atom, mob/user)
+	..()
+	var/obj/machinery/manned_turret/E = user.buckled
+	E.setDir(get_dir(E,targeted_atom))
+	user.setDir(E.dir)
+	switch(E.dir)
+		if(NORTH)
+			E.layer = BELOW_MOB_LAYER
+			user.pixel_x = 0
+			user.pixel_y = -14
+		if(NORTHEAST)
+			E.layer = BELOW_MOB_LAYER
+			user.pixel_x = -8
+			user.pixel_y = -4
+		if(EAST)
+			E.layer = ABOVE_MOB_LAYER
+			user.pixel_x = -14
+			user.pixel_y = 0
+		if(SOUTHEAST)
+			E.layer = BELOW_MOB_LAYER
+			user.pixel_x = -8
+			user.pixel_y = 4
+		if(SOUTH)
+			E.layer = ABOVE_MOB_LAYER
+			user.pixel_x = 0
+			user.pixel_y = 14
+		if(SOUTHWEST)
+			E.layer = BELOW_MOB_LAYER
+			user.pixel_x = 8
+			user.pixel_y = 4
+		if(WEST)
+			E.layer = ABOVE_MOB_LAYER
+			user.pixel_x = 14
+			user.pixel_y = 0
+		if(NORTHWEST)
+			E.layer = BELOW_MOB_LAYER
+			user.pixel_x = 8
+			user.pixel_y = -4
+	E.checkfire(targeted_atom, user)
+
+/obj/machinery/manned_turret/proc/checkfire(atom/targeted_atom, mob/user)
+	target = targeted_atom
+	if(target == user || target == get_turf(src))
+		return
+	if(world.time < cooldown)
+		if(!warned && world.time > (cooldown - cooldown_duration + rate_of_fire*number_of_shots)) // To capture the window where one is done firing
+			warned = TRUE
+			playsound(src, 'sound/weapons/sear.ogg', 100, 1)
+		return
+	else
+		cooldown = world.time + cooldown_duration
+		warned = FALSE
+		INVOKE_ASYNC(src, /obj/machinery/manned_turret/.proc/volley)
+
+/obj/machinery/manned_turret/proc/volley()
+	target_turf = get_turf(target)
+	for(var/i in 1 to number_of_shots)
+		addtimer(CALLBACK(src, /obj/machinery/manned_turret/.proc/fire_helper, target_turf), i*rate_of_fire)

```

Comment:
I think this proc used to be in the controls and didn't have the var or something